src/briefcase/platforms/windows/__init__.py

```diff
@@ -1,5 +1,191 @@
+
+    def binary_path(self, app):
+        return self.bundle_path(app) / self.packaging_root / f"{app.formal_name}.exe"
+
+    def distribution_path(self, app, packaging_format):
+        return self.platform_path / f"{app.formal_name}-{app.version}.msi"
+
+
+class WindowsCreateCommand(CreateCommand):
+    @property
+    def support_package_url_query(self):
+        """The query arguments to use in a support package query request."""
+        return [
+            ("platform", self.platform),
+            ("version", self.python_version_tag),
+            ("arch", "amd64" if (struct.calcsize("P") * 8) == 64 else "win32"),
+        ]
+
+    def output_format_template_context(self, app: BaseConfig):
+        """Additional template context required by the output format.
+
+        :param app: The config object for the app
+        """
+        # WiX requires a 3-element, integer-only version number. If a version
+        # triple isn't explicitly provided, generate one by stripping any
+        # non-numeric portions from the version number.
+        # If there are less than 3 numeric parts, 0s will be appended.
+        try:
+            version_triple = app.version_triple
+        except AttributeError:
+            parsed = parsed_version(app.version)
+            version_triple = ".".join(
+                ([str(v) for v in parsed.release] + ["0", "0"])[:3]
+            )
+
+        # The application needs a unique GUID.
+        # This is used to track the application, even if the application
+        # name changes. We can generate a default GUID using the bundle
+        # and the formal name; but you'll need to manually set this value
+        # if you ever change those two keys.
+        try:
+            guid = app.guid
+        except AttributeError:
+            # Create a DNS domain by reversing the bundle identifier
+            domain = ".".join([app.app_name] + app.bundle.split(".")[::-1])
+            guid = uuid.uuid5(uuid.NAMESPACE_DNS, domain)
+            self.logger.info(f"Assigning {app.app_name} an application GUID of {guid}")
+
+        try:
+            install_scope = "perMachine" if app.system_installer else "perUser"
+        except AttributeError:
+            # system_installer not defined in config; default to asking the user
+            install_scope = None
+
+        return {
+            "version_triple": version_triple,
+            "guid": str(guid),
+            "install_scope": install_scope,
+        }
+
+
+class WindowsRunCommand(RunCommand):
+    def run_app(self, app: BaseConfig, **kwargs):
+        """Start the application.
+
+        :param app: The config object for the app
+        """
+        self.logger.info("Starting app...", prefix=app.app_name)
+        try:
+            # Start streaming logs for the app.
+            self.logger.info("=" * 75)
+            self.subprocess.run(
+                [os.fsdecode(self.binary_path(app))],
+                check=True,
+            )
```

Comment:
I noticed that the new windows stub app output wasn't included in the briefcase log.....so, I added `stream_output=True` (and even `stderr=subprocess.STDOUT`) and I couldn't get this new output in to the briefcase log. The stub app is seemingly bypassing the pipe that python should be connecting to the app process' output....maybe windows is ignoring the pipe...?

Did you expect this? Or understand what might be happening?

Reply:
... I'm going to say some words that my mother will not be very happy to hear me say. 

Capturing stdout/err for a windows GUI app has taken me almost a week to get working *at all*. Windows GUI apps don't have a stdout/stderr by default, so we can't just redirect stdout/err. When you see the output in the terminal, you're not seeing stdout/stderr - you're seeing the output of a Windows specific device `CONOUT$`... and even then, you need to explicitly redirect stdout/err to `CONOUT$`. This is what the stub app does - if there's a console, it attaches and does the redirect; if there isn't it re-opens the stdout/err file descriptors as a file, and logs the output.

However, `subprocess.run()` doesn't have any handling for `CONOUT$`. It defines [`subprocess.STD_OUTPUT_HANDLE`](https://docs.python.org/3/library/subprocess.html#subprocess.STD_OUTPUT_HANDLE), but there doesn't appear to be any obvious way to capture this. From an initial search, [this stack overflow answer](https://stackoverflow.com/a/38749458) looks promising... if a little baroque.